constraint/pkg/types/validation.go

```diff
@@ -9,6 +9,27 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+type EvaluationEngineType string
+
+const (
+	Undefined  EvaluationEngineType = "undefined"
+	RegoEngine EvaluationEngineType = "rego"
+)
+
+type ReviewMeta struct {
```

Comment:
I like the extra level of indirection here for the following reasons:

* we may add different `metadata` things to a `ResponseMeta` and keep things somewhat modular
* we could reuse `ReviewMeta` if we wanted to add such a thing to the `Result` type itself.

Reply:
Expanding on this a bit:

> * we may add different `metadata` types to a `ResponseMeta` and keep the specific logic somewhat modular

One small papercut here is that `Response` can be returned via multiple API calls. We mostly care about the `ReviewMeta` from the `.Review()` call; if we ever need more API/ operation specific metadata, we could expose those as needed (ie `AddTemplateMeta`, etc.).